rfc.md

```diff
@@ -0,0 +1,322 @@
+    "messages": [
+      "Person was retrieved successfully"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  }
+}
+```
+
+#### Not Found Response
+```json
+{
+  "status": {
+    "messages": [
+      "No person with that name could be found"
+    ],
+    "success": false
+  }
+}
+```
+
+### /create_person 
+creates a person and returns that object with a UUID
+
+#### BODY
+```json
+{
+  "name": "The name of the person to create"
+}
+```
+
+#### Success Response
+```json
+{
+  "status": {
+    "messages": [
+      "Entry for person was successfully created"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  }
+}
+```
+
+#### Error Response (Already Exists)
+```json
+{
+  "status": {
+    "messages": [
+      "Cannot create person, someone with that name already exists."
+    ],
+    "success": false
+  }
+}
+```
+
+### /delete_person 
+marks a person and their friendships as removed
+
+#### BODY
+```json
+{
+  "uuid": "The person who you are deleting's uuid"
+}
+```
+#### Successful Response
+```json
+{
+  "status": {
+    "messages": [
+      "Person successfully removed"
+    ],
+    "success": true
+  }
+}
+```
+#### Error Response (Person doesn't exist)
+```json
+{
+  "status": {
+    "messages": [
+      "No person with that id could be found"
+    ],
+    "success": false
+  }
+}
+```
+
+#### /get_friends 
+returns a list of that person's friends
+
+#### BODY
+```json
+  {
+    "uuid": "The person whose friends you are querying's UUID"
+  }
+```
+    
+#### Successful Response
+```json
+{
+  "status": {
+    "messages": [
+      "Person successfully found"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  },
+  "friends": [
+    {
+      "uuid": "<UUID>",
+      "name": "<NAME>"
+    },
+    {
+      "uuid": "<UUID>",
+      "name": "<NAME>"
+    },
+    ...
+  ]
+}
+```
+
+#### Error Response (person cannot be found)
+```json
+{
+  "status": {
+    "messages": [
+      "Person cannot be found"
+    ],
+    "success": false
+  }
+}
+```
+
+### /create_friendship 
+creates a friendship
+
+#### BODY
+```json
+{
+  "uuids": ["first UUID", "second UUID"]
+}
+```
+
+#### Successful Response
+```json
+{
+  "status": {
+    "messages": [
+      "Friendship successfully created"
+    ],
+    "success": true
+  },
+  "person": {
+    "uuid": "<UUID>",
+    "name": "<NAME>"
+  },
+  "friends": [
+    {
+      "uuid": "<UUID>",
+      "name": "<NAME>"
+    }
+  ]
+}
+```
+
+#### Error Response (Person not found)
+```json
+{
+  "status": {
+    "messages": [
+      "Person with UUID <UUID> could not be found"
+    ],
+    "success": false
+  }
+}
+```
+
+#### Error Response (Already friends)
+```json
+{
+  "status": {
+    "messages": [
+      "These people are already friends"
+    ],
+    "success": false
+  }
+}
+```
+
+### /remove_friendship
+
+#### BODY
+```json
+{
+  "uuids": ["<UUID_1>", "<UUID_2>"]
```

Comment:
Is this uuid of the friendship? It doesn't appear to be that way given the error resp wrt to people not being friends. If it is person uuid, you probably need a "from" and "friends" uuid(s).

This appears to be one of the only requests that accepts a list of uuids.

Reply:
This is a `uuid` of the two people making up the friendship. I had the idea that friendship ids are internal implementation details, but if that's not a direction we want to go, I can rework this and the other friendship actions

Reply:
I guess then I don't understand the err response if people are not friends:

`"Friendship could not be removed, <UUID_1> and <UUID_2> are not friends"`

Reply:
The `uuid`s are of the two people that we are trying to represent as friends

Reply:
But the UUIDs in the request are a uuid of a friendship. How would that translate into an error of two people not being friends?

Reply:
That was what I was saying earlier. The uuids were of the two people, not the friendship itself. Friendships don't currently have uuids. I can change that if you think it's better

Reply:
So if the uuids here in the request are user uuids, and they are passed in as a list, what happens when I pass in 3 or more? Cartesian product?

Reply:
for now I was thinking an error, that sounds like an extension

Reply:
I can also move to two keys

Reply:
I would be explicit with the contract. Lists denote, for all intents and purposes, an unbounded set. If only two values should be passed, two keys should be used, yes.

Reply:
Will do